File: lokask-backend/backend/internal/domain/blog.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Blog struct {
	ID       uuid.UUID `db:"id" json:"id"`
	AuthorID uuid.UUID `db:"author_id" json:"author_id"`

	Title         string `db:"title" json:"title"`
	Summary       string `db:"summary" json:"summary"`
	Content       string `db:"content" json:"content"`
	CoverImageURL string `db:"cover_image_url" json:"cover_image_url"`

	City    string `db:"city" json:"city"`
	Country string `db:"country" json:"country"`

	Rating      float64 `db:"rating" json:"rating"`
	ReviewCount int     `db:"review_count" json:"review_count"`

	CreatedAt time.Time `db:"created_at" json:"created_at"`
	UpdatedAt time.Time `db:"updated_at" json:"updated_at"`

	// Fields for "JOIN" queries (not in the blogs table, but useful for UI)
	AuthorName   string `db:"author_name" json:"author_name,omitempty"`
	AuthorAvatar string `db:"author_avatar" json:"author_avatar,omitempty"`
}

File: asklocal/mobile/lib/models/blog.dart
class Blog {
  final String id;
  final String authorId;
  final String title;
  final String summary;
  final String content;
  final String coverImageUrl;
  final String city;
  final String authorName;
  final String authorAvatar;
  final DateTime createdAt;

  final double rating;
  final int reviewCount;

  Blog({
    required this.id,
    required this.authorId,
    required this.title,
    required this.summary,
    required this.content,
    required this.coverImageUrl,
    required this.city,
    required this.authorName,
    required this.authorAvatar,
    required this.createdAt,
    required this.rating,
    required this.reviewCount,
  });

  factory Blog.fromJson(Map<String, dynamic> json) {
    return Blog(
      id: json['id'] ?? "",
      authorId: json['author_id'] ?? "",
      title: json['title'] ?? "Untitled",
      summary: json['summary'] ?? "",
      content: json['content'] ?? "",
      coverImageUrl: json['cover_image_url'] ?? "",
      city: json['city'] ?? "",
      authorName: json['author_name'] ?? "Unknown",
      authorAvatar: json['author_avatar'] ?? "",
      createdAt: DateTime.tryParse(json['created_at'] ?? "") ?? DateTime.now(),
      rating: (json['rating'] ?? 0).toDouble(),
      reviewCount: json['review_count'] ?? 0,
    );
  }
}
